exporter/awsxrayexporter/internal/translator/segment.go

```diff
@@ -107,9 +113,23 @@ func MakeSegment(span ptrace.Span, resource pcommon.Resource, indexedAttrs []str
 
 	// X-Ray segment names are service names, unlike span names which are methods. Try to find a service name.
 
+	// support x-ray specific service name attributes as segment name if it exists
+	if span.Kind() == ptrace.SpanKindServer || span.Kind() == ptrace.SpanKindConsumer {
+		if localServiceName, ok := attributes.Get(awsLocalService); ok {
+			name = localServiceName.Str()
+		}
+	}
+	if span.Kind() == ptrace.SpanKindClient || span.Kind() == ptrace.SpanKindProducer {
+		if remoteServiceName, ok := attributes.Get(awsRemoteService); ok {
+			name = remoteServiceName.Str()
+		}
+	}
+
 	// peer.service should always be prioritized for segment names when set because it is what the user decided.
```

Comment:
This comment will now be false since the segment name priority will be `aws.local.service` or `aws.remote.service` value. Correct?

Also, seems like a breaking behavior to me to change the segment naming.

Reply:
Good catches, will update the comments. 

The new segment naming behavior will only be applied to the new aws apm users as the new aws service attribute will not be populated in trace spans unless aws customers opt-in to enable it.